src/components/Overlay/Overlay.tsx

```diff
@@ -10,7 +10,7 @@ interface OverlayProps {
 export const Overlay: FC<OverlayProps> = ({children, display, closeOverlay}) => {
   const handleClick = (e: MouseEvent) => {
     // Po kliknutí na overlay, mimo akýchkoľvek iných elementov, sa overlay zatvorí.
-    if ((e.target as Element).classList.contains('overlay')) {
+    if ((e.target as Element).classList[0]?.includes('overlay')) {
```

Comment:
este mi napadlo... asi sme tu mohli pouzit `styles.overlay`, to by mal byt presne ten classname vlastne 😄